Issue 3547 ratelimit per host

iamargus95 opened the pull request:
## Proposed changes
- Add `-rlh` flag to options.
- Add ratelimit to executeHostSpray using value set in flag.

## Checklist
- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [x] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

iamargus95 commented:
Hi @tarunKoyalwar,
- I've created this draft PR to address the [issue-3547](https://github.com/projectdiscovery/nuclei/issues/3547).
- I've added `TODO:` comments near places where I need some clarity.

iamargus95 commented:
@ehsandeep 
The default for the `-rlh` flag in this PR is 150, Should it be something else like 99..9 so as to not limit it when no value is provided?
CC: @tarunKoyalwar

tarunKoyalwar commented:
@iamargus95  , i think we can set default to `60` i.e 60 req / min . since this is average of what bug bounty programs allow

iamargus95 commented:
@tarunKoyalwar
Updated the PR
- Changed default value of `-rlh` flag to 60 requests per minute. 
- CMIIW : Stopping the ratelimiter after the `wp.Wait()`.
     - Couldn't find a good example of the ratelimit library. 😅 
- There are some comments (Marked as TODO:) regarding the flag description I'd like some help on as well.

tarunKoyalwar commented:
Hii @iamargus95 thanks for the PR , but as you can tell implementing ratelimit per host in nuclei is difficult task . your current implementation adds ratelimit for templates . but a template can have/send more than 1 request (ex: race,payloads etc) . i.e why ratelimit needs to be internally at request level in protocols . and needs to be abstracted for both scan-strategies

I'm closing this PR since the required implementation is different and this issue needs more context about nuclei internals so I will be working on this issue in a seperate PR . If you have any suggestions on issue feel free to add them in issue
